### PR TITLE
fix(auth/oauth): route oauth to auth and config changes

### PR DESCRIPTION
**Pull request by jrgm**
I don't know if this is complete, but it gets things nominally working.

r? - @dannycoates, @vladikoff, @6a68

### CODE REVIEW

File: aws/environments/jrgm.yml [hunk 1]
```diff
@@ -10,6 +10,6 @@ ec2_volume_size: 24
 owner: "dev-fxacct@mozilla.org"
 reaper_spare_me: "true"
 
-fxadev_git_version: fix-subscription-config
+fxadev_git_version: docker-jrgm-fix-auth-oauth
```
Comment (jrgm) [inline aws/environments/jrgm.yml:13]: You can safely ignore this.

**Comment by jrgm** [comment]
Hrm. I think that's the wrong branch, or a needed commit is not on that branch. I'll find it (alters nginx routing).

**Comment by jrgm** [comment]
Okay, I added the left behind commits to this branch. Please take a look.

File: roles/content/templates/nginx.conf.j2 [hunk 2]
```diff
@@ -2,6 +2,7 @@ location / {
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
     proxy_set_header Host $http_host;
     proxy_redirect off;
+    resolver 172.31.0.2 valid=30s;
```
Comment (jrgm) [inline roles/content/templates/nginx.conf.j2:5]: That magic IP is +2 to the bottom of the vpc we're using in us-west-2. I should have parameterized this, but didn't.

Comment (jrgm) [reply roles/content/templates/nginx.conf.j2:5]: https://ryangilliganblog.wordpress.com/2018/09/19/aws-nginx-returns-499-response-errors-when-using-downstream-load-balancer/

File: roles/oauth/templates/upstream.conf.j2 [hunk 3]
```diff
@@ -9,10 +9,13 @@ server {
...
 
+  # Somthing that arrives as https://oauth-{{name}}.dev.lcip.org/
+
   location / {
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
     proxy_set_header Host $http_host;
     proxy_redirect off;
+    rewrite ^/(.*)$ /auth/$1 break;
     proxy_pass http://upstream_oauth_server;
   }
```
Comment (jrgm) [inline roles/oauth/templates/upstream.conf.j2:20]: For fxa-dev, all the hostnames already resolve to the same "host". It's nginx rules that decide where things go, and the above send "oauth" requets to the auth daemon. I think.

File: roles/auth/tasks/main.yml [hunk 4]
```diff
@@ -46,6 +51,15 @@
     mozilla/fxa-auth-server node scripts/gen_vapid_keys.js
     creates=/data/config/auth/vapid-keys.json
 
+- name: generate oauth/auth openid key file, and dummy oldkeyfile
+  become: true
+  command: >-
+    docker run --rm --net=host -v /data/config/openid:/openid
+      -e FXA_OPENID_KEYFILE=/openid/openid-key.json
+      -e FXA_OPENID_OLDKEYFILE=/openid/openid-old-key.json
+    mozilla/fxa-auth-server node fxa-oauth-server/scripts/gen_keys.js
```
Comment (dannycoates) [inline roles/auth/tasks/main.yml]: I just deleted fxa-oauth-server from master today. the new location for this script is `scripts/oauth_gen_keys.js`

**Comment by dannycoates** [comment]
Once stable has train-150 it should be safe to remove oauth-server completely

**Review by dannycoates — APPROVED** [review]
Thanks @jrgm

**Pull request merged**